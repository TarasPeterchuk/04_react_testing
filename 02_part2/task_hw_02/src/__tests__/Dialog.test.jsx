import React from 'react';
import { shallow } from 'enzyme';
import Dialog from '../Dialog';

const mockClose = jest.fn();

describe('Dialog', () => {
  it('should not display dialog', () => {
    const wrappedComponent = shallow(<Dialog isOpen={false} />);
    expect(wrappedComponent.find('.dialog').exists()).toEqual(false);
  });

  it('should display dialog', () => {
    const wrappedComponent = shallow(<Dialog isOpen />);
    expect(wrappedComponent.find('.dialog').exists()).toBeTruthy();
  });

  it('should close dialog after click close button', () => {
    const wrappedComponent = shallow(<Dialog isOpen onClose={mockClose} />);
    wrappedComponent.find('.dialog__close-btn').simulate('click');
    expect(mockClose).toBeCalled();
  });

  it('should display title', () => {
    const testTitle = 'Profile Facebook';
    const wrappedComponent = shallow(<Dialog isOpen title={testTitle} />);
    expect(wrappedComponent.find('.dialog__title').text()).toEqual(testTitle);
  });

  it('should display Dialog section', () => {
    const wrappedComponent = shallow(<Dialog isOpen />);
    expect(wrappedComponent.find('.dialog')).toMatchSnapshot();
  });
});
