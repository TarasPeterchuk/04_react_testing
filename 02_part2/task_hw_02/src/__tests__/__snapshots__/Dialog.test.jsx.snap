// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should display Dialog section 1`] = `
<div
  className="dialog"
>
  <div
    className="dialog__heading"
  >
    <h4
      className="dialog__title"
    />
    <button
      className="dialog__close-btn"
    >
      +
    </button>
  </div>
  <div
    className="dialog__content"
  />
</div>
`;

exports[`Dialog should initialize counter with 0 1`] = `
<div
  className="dialog"
>
  <div
    className="dialog__heading"
  >
    <h4
      className="dialog__title"
    />
    <button
      className="dialog__close-btn"
    >
      +
    </button>
  </div>
  <div
    className="dialog__content"
  />
</div>
`;
